assets/js/state/sagas/activityLogsSettings.test.js

```diff
@@ -0,0 +1,96 @@
+import { recordSaga } from '@lib/test-utils';
+
+import { networkClient } from '@lib/network';
+import MockAdapter from 'axios-mock-adapter';
+
+import { activityLogsSettingsFactory } from '@lib/test-utils/factories/activityLogsSettings';
+
+import {
+  startLoadingActivityLogsSettings,
+  setActivityLogsSettings,
+  setActivityLogsSettingsErrors,
+  setEditingActivityLogsSettings,
+  setNetworkError,
+} from '@state/activityLogsSettings';
+
+import {
+  fetchActivityLogsSettings,
+  updateActivityLogsSettings,
+} from './activityLogsSettings';
+
+describe('Activity Logs Settings saga', () => {
+  describe('Fetching Activity Logs Settings', () => {
+    it('should successfully fetch activity logs settings', async () => {
+      const axiosMock = new MockAdapter(networkClient);
+      const successfulResponse = activityLogsSettingsFactory.build();
+
+      axiosMock.onGet('/settings/activity_log').reply(200, successfulResponse);
+
+      const dispatched = await recordSaga(fetchActivityLogsSettings);
+
+      expect(dispatched).toEqual([
+        startLoadingActivityLogsSettings(),
+        setActivityLogsSettings(successfulResponse),
+      ]);
+    });
+
+    it.each([403, 404, 500, 502, 504])(
+      'should put a network error flag on failed fetching',
+      async (status) => {
+        const axiosMock = new MockAdapter(networkClient);
+        axiosMock.onGet('/settings/activity_logs').reply(status);
+
+        const dispatched = await recordSaga(fetchActivityLogsSettings);
+
+        expect(dispatched).toEqual([
+          startLoadingActivityLogsSettings(),
+          setNetworkError(true),
```

Comment:
There is a typo in the path of the request
```js
// current, incorrect. Always results in 404
axiosMock.onGet('/settings/activity_logs').reply(status);

// correct, without the trailing `s`
axiosMock.onGet('/settings/activity_log').reply(status);
```
---
Since we have this check in `fetchActivityLogsSettings`
```js
if (errorCode !== 403) {
  yield put(setNetworkError(true));
}
```
I was expecting the `403` scenario to **not** result in `setNetworkError(true)` being part of `dispatched`.

This was driving me crazy :sweat_smile:

Reply:
done in https://github.com/trento-project/web/pull/2727/commits/52303cac574bda7f2b54edbb8574cd63f3089ca2